lib/chef/provider/package/chocolatey.rb

```diff
@@ -152,11 +147,15 @@ def check_resource_semantics!
         #
         # @return [String] full path of choco.exe
         def choco_exe
-          @choco_exe ||= ::File.join(
-            choco_install_path,
-            "bin",
-            "choco.exe"
-            )
+          @choco_exe ||=
+              # if this check is in #define_resource_requirements, it won't get
+              # run before choco.exe gets called from #load_current_resource.
+              exe_path = ::File.join(choco_install_path.to_s, "bin", "choco.exe")
+          if !::File.exist?(exe_path)
```

Comment:
looks like some indentation issues here. Also `unless` may be clearer than `if !`.